java/tools/src/main/java/org/apache/tsfile/tools/TsFileTool.java

```diff
@@ -188,6 +202,7 @@ private static Tablet genTablet(
       map.put(column.name, i);
     }
     try {
+      sortByColumnIndex(lineList);
       for (int i = 0; i < num; i++) {
         String line = lineList.get(i);
         String[] lineArray = line.split(schema.separator);
```

Comment:
The lines are split twice, which is unnecessary.
You may first split them, sort them, and then iterate over them.